lib/eosknowledgeprivate/ekn-media-bin.c

```diff
@@ -113,7 +115,11 @@ typedef struct
   guint timeout_id;          /* Autohide timeout source id */
   gint  timeout_count;       /* Autohide timeout count since last move event */
 
-  guint tick_id;             /* Widget frame clock tick callback (used to update UI) */
+  guint   tick_id;           /* Widget frame clock tick callback (used to update UI) */
+  guint64 tick_start;
+  guint   playback_time;
+  guint   frames_rendered;   /* Total frames "rendered" */
+  guint   frames_window;     /* Frames "rendered" in the last 4s window */
```

Comment:
Here you say you're using a 4 seconds window, but below you check every 2 seconds.